smithy-aws-protocol-tests/model/rpcV2/cbor-structs.smithy

```diff
@@ -0,0 +1,322 @@
+
+use smithy.protocols#rpcv2Cbor
+use smithy.test#httpRequestTests
+use smithy.test#httpResponseTests
+
+
+@httpRequestTests([
+    {
+        id: "RpcV2CborSimpleScalarProperties",
+        protocol: rpcv2Cbor,
+        documentation: "Serializes simple scalar properties",
+        headers: {
+            "smithy-protocol": "rpc-v2-cbor",
+            "Accept": "application/cbor",
+            "Content-Type": "application/cbor"
+        }
+        method: "POST",
+        bodyMediaType: "application/cbor",
+        uri: "/service/aws.protocoltests.rpcv2Cbor.RpcV2Protocol/operation/SimpleScalarProperties",
+        body: "v2lieXRlVmFsdWUFa2RvdWJsZVZhbHVl+z/+OVgQYk3TcWZhbHNlQm9vbGVhblZhbHVl9GpmbG9hdFZhbHVl+kDz989saW50ZWdlclZhbHVlGQEAaWxvbmdWYWx1ZRkmkWpzaG9ydFZhbHVlGSaqa3N0cmluZ1ZhbHVlZnNpbXBsZXB0cnVlQm9vbGVhblZhbHVl9f8="
+        params: {
+            trueBooleanValue: true,
+            falseBooleanValue: false,
+            byteValue: 5,
+            doubleValue: 1.889,
+            floatValue: 7.624,
+            integerValue: 256,
+            shortValue: 9898,
+            longValue: 9873,
+            stringValue: "simple"
+        }
+    },
+    {
+        id: "RpcV2CborClientDoesntSerializeNullStructureValues",
+        documentation: "RpcV2 Cbor should not serialize null structure values",
+        protocol: rpcv2Cbor,
+        method: "POST",
+        uri: "/service/aws.protocoltests.rpcv2Cbor.RpcV2Protocol/operation/SimpleScalarProperties",
+        body: "v/8=",
+        bodyMediaType: "application/cbor",
+        headers: {
+            "smithy-protocol": "rpc-v2-cbor",
+            "Accept": "application/cbor",
+            "Content-Type": "application/cbor"
+        }
+        params: {
+            stringValue: null
+        },
+        appliesTo: "client"
+    },
+    {
+        id: "RpcV2CborServerDoesntDeSerializeNullStructureValues",
+        documentation: "RpcV2 Cbor should not deserialize null structure values",
+        protocol: rpcv2Cbor,
+        method: "POST",
+        uri: "/service/aws.protocoltests.rpcv2Cbor.RpcV2Protocol/operation/SimpleScalarProperties",
+        body: "v2tzdHJpbmdWYWx1Zfb/",
+        bodyMediaType: "application/cbor",
+        headers: {
+            "smithy-protocol": "rpc-v2-cbor",
+            "Accept": "application/cbor",
+            "Content-Type": "application/cbor"
+        }
+        params: {},
+        appliesTo: "server"
+    },
+    {
+        id: "RpcV2CborSupportsNaNFloatInputs",
+        protocol: rpcv2Cbor,
+        documentation: "Supports handling NaN float values.",
+        headers: {
+            "smithy-protocol": "rpc-v2-cbor",
+            "Accept": "application/cbor",
+            "Content-Type": "application/cbor"
+        }
+        method: "POST",
+        bodyMediaType: "application/cbor",
+        uri: "/service/RpcV2Protocol/operation/SimpleScalarProperties",
+        body: "v2tkb3VibGVWYWx1Zft/+AAAAAAAAGpmbG9hdFZhbHVl+n/AAAD/"
```

Comment:
Worth noting that different implementations may expose different values for general NaN. We've encoded f64 NaN here as `7ff8000000000000` but go has defined it as `0x7ff8000000000001`, both technically NaN but fails a bitwise comparison. Not sure if we care, though. We should probably be recommending that people do document-based equality checks for these comparisons.